automated install/basic-install.sh

```diff
@@ -1977,6 +1977,102 @@ accountForRefactor() {
     fi
 }
 
+# A function that validates the pihole_setupVars.conf required for unattended installation and updates.
+validate_setupVars () {
+    printf "  Validating setupVars.conf\\n"
+    echo ""
+    # Replace any crlf with nl if they exists in the source file
+    sed -i 's/\r//'  ${setupVars}
+    # Does setupVars contain the minimume variables?
+    local requiredVars=(PIHOLE_INTERFACE IPV4_ADDRESS PIHOLE_DNS_1 PIHOLE_DNS_2)
+    # Variables with true or false as the only options
+    local optionalVarsTF=(QUERY_LOGGING INSTALL_WEB_SERVER INSTALL_WEB_INTERFACE BLOCKING_ENABLED DNS_FQDN_REQUIRED DNS_BOGUS_PRIV DNSSEC CONDITIONAL_FORWARDING)
+    source ${setupVars}
+    for var in "${requiredVars[@]}"; do
+        if [[ -z "${!var}" ]]; then
+            echo "${var} is not set"
+            exit 1
+        else
+            echo "${var} is ${!var}"
+        fi
+    done
+    for i in "${optionalVarsTF[@]}"; do
+        if [[ "${!var}" == "true" ]] || [[ "${!var}" == "false" ]]; then
+            echo "${var} is ${!var}"
+        elif [[ -z "${!var}" ]]; then
+            echo "${var} is not set"
+        else
+            echo "${var} is ${!var}"
+            echo "If ${var} is set in setupVars.conf, valid options are: true or false."
+            exit 1
+        fi
+    done
+    # INSTALL_WEB_SERVER must be set to true if INSTALL_WEB_INTERFACE is set to true
+    if [[ "${INSTALL_WEB_INTERFACE}" == true ]] && [[ "${INSTALL_WEB_SERVER}" == false ]]; then
+        echo "You can install the Web Server without the web interface, but you cannot install the web interface without the web server."
```

Comment:
As of above, installing a web server without the web interface does not make any sense, but installing the web interface but allowing to install a different web server or Lighttpd with custom PHP handler is a common use case and the reason why this variable + installer option has been added.